### PR TITLE
React on github release

### PR DESCRIPTION
**Pull request by phracek**
Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>
This pull request adds support for syncing upstream releases into dist-git.
Based on the `.packit.yaml` the configuration file, user can select either creating PR or directly push changes into a dist-git branch.

- [x] packit-service is able to sync releases from GitHub event
- [x] test is covered

### CODE REVIEW

**Comment by phracek** [comment]
I have tested locally and try to push it into dist-git but it failed with expected traceback.
~~~
[2019-07-18 10:31:03,699: DEBUG/ForkPoolWorker-1] Starting new HTTPS connection (1): src.fedoraproject.org:443
[2019-07-18 10:31:04,295: DEBUG/ForkPoolWorker-1] https://src.fedoraproject.org:443  "HEAD /lookaside/pkgs/packit/packitos-0.4.2.tar.gz/ HTTP/1.1" 200 0
[2019-07-18 10:31:04,340: INFO/ForkPoolWorker-1] Archive packitos-0.4.2.tar.gz found in lookaside cache (skipping upload).
[2019-07-18 10:31:04,341: DEBUG/ForkPoolWorker-1] Upstream archive name is 'packitos-0.4.2.tar.gz'
[2019-07-18 10:31:04,341: DEBUG/ForkPoolWorker-1] About to add all & commit
[2019-07-18 10:31:04,342: DEBUG/ForkPoolWorker-1] Popen(['git', 'add', '-A'], cwd=/tmp/tmpayalp2y3, universal_newlines=False, shell=None)
[2019-07-18 10:31:04,352: DEBUG/ForkPoolWorker-1] Popen(['git', 'diff', '--cached', '--abbrev=40', '--full-index', '--raw'], cwd=/tmp/tmpayalp2y3, universal_newlines=False, shell=None)
[2019-07-18 10:31:04,362: DEBUG/ForkPoolWorker-1] Popen(['git', 'diff', '--abbrev=40', '--full-index', '--raw'], cwd=/tmp/tmpayalp2y3, universal_newlines=False, shell=None)
[2019-07-18 10:31:04,372: DEBUG/ForkPoolWorker-1] Popen(['git', 'checkout', 'master'], cwd=/sandcastle, universal_newlines=False, shell=None)
[2019-07-18 10:31:04,742: DEBUG/MainProcess] pidbox received method enable_events() [reply_to:None ticket:None]
[2019-07-18 10:31:04,945: INFO/ForkPoolWorker-1] cleaning up the mess
[2019-07-18 10:31:04,946: DEBUG/ForkPoolWorker-1] Initialize kubernetes client
[2019-07-18 10:31:04,946: INFO/ForkPoolWorker-1] loading incluster config
[2019-07-18 10:31:04,960: INFO/ForkPoolWorker-1] removing sandbox pod
[2019-07-18 10:31:04,976: DEBUG/ForkPoolWorker-1] response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods \"docker-io-usercont-sandcastle-20190718-103104946292\" not found","reason":"NotFound","details":{"name":"docker-io-usercont-sandcastle-20190718-103104946292","kind":"pods"},"code":404}

[2019-07-18 10:31:04,976: DEBUG/ForkPoolWorker-1] (404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Cache-Control': 'no-store', 'Content-Type': 'application/json', 'Date': 'Thu, 18 Jul 2019 10:31:04 GMT', 'Content-Length': '270'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods \"docker-io-usercont-sandcastle-20190718-103104946292\" not found","reason":"NotFound","details":{"name":"docker-io-usercont-sandcastle-20190718-103104946292","kind":"pods"},"code":404}


[2019-07-18 10:31:04,976: DEBUG/ForkPoolWorker-1] remove contents of the PV
[2019-07-18 10:31:04,981: INFO/ForkPoolWorker-1] volume was not empty
[2019-07-18 10:31:04,981: DEBUG/ForkPoolWorker-1] content of the volume: [PosixPath('/sandcastle/LICENSE'), PosixPath('/sandcastle/CHANGELOG.md'), PosixPath('/sandcastle/.pre-commit-config.yaml'), PosixPath('/sandcastle/Dockerfile'), PosixPath('/sandcastle/packit.spec'), PosixPath('/sandcastle/Makefile'), PosixPath('/sandcastle/release-conf.yaml'), PosixPath('/sandcastle/tests'), PosixPath('/sandcastle/.git_archival.txt'), PosixPath('/sandcastle/packit'), PosixPath('/sandcastle/docs'), PosixPath('/sandcastle/README.md'), PosixPath('/sandcastle/setup.py'), PosixPath('/sandcastle/.gitignore'), PosixPath('/sandcastle/CONTRIBUTING.md'), PosixPath('/sandcastle/fedora-tests'), PosixPath('/sandcastle/.packit.yaml'), PosixPath('/sandcastle/ansible.cfg'), PosixPath('/sandcastle/.gitattributes'), PosixPath('/sandcastle/Jenkinsfile'), PosixPath('/sandcastle/Vagrantfile'), PosixPath('/sandcastle/files'), PosixPath('/sandcastle/setup.cfg'), PosixPath('/sandcastle/.git'), PosixPath('/sandcastle/recipe-tests.yaml')]

[2019-07-18 10:31:05,244: ERROR/ForkPoolWorker-1] Task task.steve_jobs.process_message[f2934ca6-ddac-47fb-8674-8958a01eadd0] raised unexpected: PackitException('No changes are present in the dist-git repo: nothing to commit.')
--



~~~

**Comment by TomasTomecek** [comment]
I'm probably blind but I can't see any traceback nor an obvious error in that output.

File: packit_service/worker/jobs.py [hunk 1]
```diff
@@ -311,21 +311,24 @@ def process_jobs(
...
-                        commit_sha = nested_get(event, "pull_request", "head", "sha")
-                        r = BuildStatusReporter(project, commit_sha)
-                        msg = "Account is not whitelisted!"
-                        r.report("failure", msg, url=FAQ_URL)
-
-                        handlers_results[job.job.value] = HandlerResults(
-                            success=False, details={"msg": msg}
-                        )
-                        return handlers_results
+                        if trigger == JobTriggerType.pull_request:
```
Comment (TomasTomecek) [inline packit_service/worker/jobs.py]: So, this pretty much means that I'd be able to create a new github project, install packit service, and start committing to fedora dist-git wherever I wanted. Without being approved.

@rpitonak will you fix that in your PR or should it be fixed here?

Comment (phracek) [reply packit_service/worker/jobs.py]: ohh, you are right, the condition can be moved only to `commit_sha` row. If we get rid of commit_sha, then no condition is needed.

**Comment by phracek** [comment]
> I'm probably blind but I can't see any traceback nor an obvious error in that output.

It was not copied for some reason, I have updated it.

File: packit_service/worker/jobs.py (hunk 1 above)
Comment (rpitonak) [reply packit_service/worker/jobs.py]: @TomasTomecek Yes, it is going to check whitelist for both triggers `release` and `pull_request` and then reports status just for the PR.

**Comment by phracek** [comment]
Whi the lastest commit https://github.com/packit-service/packit-service/pull/51/commits/754b9662fae41c34b21ae2080f0207b6acf82408 if the account is not whitelisted False HandlerResult is returned.

**Pull request merged**